### PR TITLE
reset ure form to original values when cancelled

### PR DESCRIPTION
**Pull request by akash1810**
GIFs may not illustrate the bug. Steps to reproduce:
- Open the usage rights form for any image
- Make a change but don't save, instead press cancel to close the form
- Open the usage rights form again, and the value of the form is that of the change from above rather than the original.

# Before
ure-reset-on-cancel

# After
ure-reset-on-cancel-after

I've also refactored to remove a few unused functions and move instantiation of controller properties higher.

### CODE REVIEW

File: kahuna/public/js/usage-rights/usage-rights-editor.js [hunk 1]
```diff
@@ -187,6 +183,8 @@ usageRightsEditor.controller(
     function uiError(error) {
         ctrl.error = error.body.errorMessage;
     }
+
+    init();
```
Comment (jamesgorrie) [inline kahuna/public/js/usage-rights/usage-rights-editor.js]: This means we will be doing a call to the API twice which isn't needed.

We could either memoize the categories in the service, or have two methods here, `fetchCategories` & `setCategory`.

**Comment by theefer** [comment]
Suggested to @akash1810 to make the ure conditional behind an `ng:if` instead of `ng:show`. That should have the benefit of killing the DOM and controller when the dialog is closed, and re-initialised it fully every time it gets opened. That way the dialog doesn't have to be aware of some kind of "reset" when hidden and shown. This is slightly more expensive from a JS/DOM POV, but makes the logic much simpler and less stateful.

**Comment by jamesgorrie** [comment]
@theefer :+1::fireworks:

**Comment by akash1810** [comment]
Updated. `ng:if` is so much nicer!

File: kahuna/public/js/usage-rights/usage-rights-editor.js (hunk 1 above)
Comment (akash1810) [inline kahuna/public/js/usage-rights/usage-rights-editor.js]: > We could either memoize the categories in the service

The service already does this - https://github.com/guardian/grid/blob/master/kahuna/public/js/services/api/edits-api.js#L13

**Comment by jamesgorrie** [comment]
:+1:

**Pull request merged**